lib/internal/process/stdio.js

```diff
@@ -8,26 +8,64 @@ function setupStdio() {
   function getStdout() {
     if (stdout) return stdout;
     stdout = createWritableStdioStream(1);
+
+    Object.defineProperty(stdout._writableState, 'ended', {
+      set() {},
+      get() {
+        return false;
+      }
+    });
+    // XXX(Fishrock123): This should work I think, but it causes
+    // problems with some child-process related tests.
+    //
+    // Object.defineProperty(stdout, 'destroyed', {
+    //   set() {},
+    //   get() {
+    //     return true;
+    //   }
+    // });
+
     stdout.destroy = stdout.destroySoon = function(er) {
+      stdout.destroyed = true;
       er = er || new Error('process.stdout cannot be closed.');
       stdout.emit('error', er);
     };
     if (stdout.isTTY) {
       process.on('SIGWINCH', () => stdout._refreshSize());
     }
+    stdout.on('error', () => {});
```

Comment:
Hmmm, perhaps. I'd like to have a bit more discussion, I don't know if throwing from stdio on process is a good idea either. There are a lot of libraries that do raw writes that could be affected by e.g. `EPIPE` from an outside source, and that doesn't really make sense to crash on IMO.